### PR TITLE
New Test scripts for ResyncFailedPoolOutOfRebalance

### PR DESCRIPTION
**Pull request by vprabhakar-px**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: tests/basic/storage_pool_test.go [hunk 1]
```diff
@@ -7923,3 +7923,94 @@ var _ = Describe("{DiffPoolExpansionFromMaintenanceNode}", func() {
...
+			dash.VerifyFatal(resizeErr, nil,
+				fmt.Sprintf("Verify pool %s on expansion using auto option", poolUUID))
+		}
+		log.InfoD("Validate pool rebalance after drive add")
+		err = ValidatePoolRebalance(*nodeDetail, poolToBeResized.ID)
+
+		// Validate Volume resync if any volume got in to resync mode
+		for _, eachContext := range contexts {
+			vols, err := Inst().S.GetVolumes(eachContext)
+			log.FailOnError(err, "Failed to get volumes from context")
```
Comment (lsrinivas-pure) [inline tests/basic/storage_pool_test.go:7990]: can you add check if vol has already repl3 , if so you can make it 2 and make it 3 again.
Also you can make this operation only on volume which has "poolUUID" as one of its replica

File: tests/common.go [hunk 2]
```diff
@@ -1726,14 +1726,15 @@ func GetStorageNodes() ([]node.Node, error) {
...
-
-	for _, node := range nodes {
-		devices, err := Inst().V.GetStorageDevices(node)
-		if err != nil {
-			return nil, err
-		}
-		if len(devices) > 0 {
-			storageNodes = append(storageNodes, node)
+	for _, eachNode := range nodes {
+		if node.IsStorageNode(eachNode) {
```
Comment (lsrinivas-pure) [inline tests/common.go:1730]: this seems like a redundant method, can you delete this and replace it with node.GetStorageNodes

Comment (vprabhakar-px) [reply tests/common.go:1730]: This is not redundant , I am modifying already available function , which is GetStorageNodes() , 
I am basically checking if the node is storageNode , and then proceeding

File: tests/basic/storage_pool_test.go (hunk 1 above)
Comment (vprabhakar-px) [reply tests/basic/storage_pool_test.go:7990]: done

File: drivers/node/node_registry.go [hunk 3]
```diff
@@ -217,3 +217,13 @@ func GetNodeDetailsByNodeName(nodeName string) (Node, error) {
 	}
 	return Node{}, fmt.Errorf("failed to get Node Details by Node Name [%s] ", nodeName)
 }
+
+// GetNodeUuidByName get node uuid for a given node name
+func GetNodeUuidByName(nodeName string) (string, error) {
+	for _, n := range nodeRegistry {
+		if n.Name == nodeName {
+			return n.uuid, nil
```
[the remark this answers is not in the document]
Comment (vprabhakar-px) [reply drivers/node/node_registry.go]: already deleted

**Review by lsrinivas-pure — APPROVED** [review]
lgtm

**Pull request merged**